### PR TITLE
Add fix for type aliasing issue with get_user_model() assignement and mypy

### PR DESCRIPTION
**Pull request by jhurej**

### CODE REVIEW

**Comment by jhurej** [comment]
Probably I will revert drf downgrade becuase of this PR https://github.com/merixstudio/django-trench/pull/162

**Review by wmaciejewskimer — APPROVED** [review]

File: trench/backends/application.py [hunk 1]
```diff
@@ -11,7 +12,7 @@
 from trench.settings import trench_settings
 
 
-User = get_user_model()
+User: settings.AUTH_USER_MODEL = get_user_model()
```
Comment (allow-cookies) [inline trench/backends/application.py]: `settings.AUTH_USER_MODEL` doesn't seem like a proper type definition. I'm aware that `mypy` doesn't report an issue after adding this value as a type hint but it's not a valid argument as you can add:

```python
User: settings.WHATEVER = get_user_model()
```

and `mypy` will report that everything is fine as well.

The `django.contrib.auth.models.AbstractUser` would be a better fit here as a type hint.

**Review by allow-cookies — CHANGES_REQUESTED** [review]

File: trench/backends/application.py (hunk 1 above)
Comment (wmaciejewskimer) [reply trench/backends/application.py]: Good point, fixed in new commit

**Review by allow-cookies — APPROVED** [review]
Looks good 👍

**Pull request merged**